### ISSUE TITLE
Errors when setting `noccb = 0`

### ISSUE THREAD

**Issue by shufay** [linusjoonho/ipie#234]
Hi, 

When I set the number of spin down electrons `noccb = 0`, I'm encountering errors associated with arrays having zero-length along certain dimensions. This error occurs in the method `trial_wavefunction.single_det.SingleDet.calculate_energy`. I attach the python script `run_afqmc_error.py` which constructs the Hamiltonian and trial wave function using 1-body and 2-body integrals, MO coefficients, and Choleskies obtained from a previous HF calculation (HDF5 file too large to upload; can send in some other way?). I also attach the output from `stderr` and `stdout`. The `develop` branch was used.

There might be an error also due to `noccb = 0` associated with undefined variables in `estimators.utils.gab_spin`.

[error.zip](https://github.com/linusjoonho/ipie/files/11620161/error.zip)

**Comment by fdmalone** [comment]
Thanks for raising this. I have a feeling everything worked for fully spin-polarised systems at some point in the past so I imagine it's a small fix.

**Comment by shufay** [comment]
Sure, I can take a look--thanks!

**Comment by fdmalone** [comment]
In particular, I think the main problem is with gab_spin. I think bypassing the green's function call and returning zero arrays with one of the dimensions as size zero then everything might work?

**Comment by shufay** [comment]
Just an update: I made some minor changes [here](https://github.com/shufay/ipie/commit/eee7c57581158b2eed21514e33a0dbddddedf405) to mitigate the errors. However, I'm now encountering errors associated with MPI:
```
energy = [-0.22051544+1.41330648e-18j -0.09321428+1.36815212e-18j
 -0.12730116+4.51543597e-20j]
# random seed is 59306159
# WARNING: skipping trial energy calculation is requested.
# Available memory on the node is 8.000 GB
            Block                   Weight            WeightFactor            HybridEnergy                  ENumer                  EDenom                  ETotal                  E1Body                  E2Body
                0   6.4000000000000000e+02  6.4000000000000000e+02  0.0000000000000000e+00 -1.4112988395608110e+02  6.4000000000000000e+02 -2.2051544368137674e-01 -9.3214281868849627e-02 -1.2730116181252707e-01
Traceback (most recent call last):
  File "/Users/shufay/Documents/in_prep/moire/ginsburg/ipie/run_afqmc.py", line 167, in <module>
    afqmc.run(comm=comm)
  File "/Users/shufay/Documents/ipie/ipie/qmc/afqmc.py", line 323, in run
    self.pcontrol.pop_control(self.walkers, comm)
  File "/Users/shufay/Documents/ipie/ipie/walkers/pop_controller.py", line 115, in pop_control
    pair_branch(walkers, comm, self.max_weight, self.min_weight, self.timer)
  File "/Users/shufay/Documents/ipie/ipie/walkers/pop_controller.py", line 329, in pair_branch
    comm.Gather(
  File "mpi4py/MPI/Comm.pyx", line 706, in mpi4py.MPI.Comm.Gather
mpi4py.MPI.Exception: MPI_ERR_TRUNCATE: message truncated
```

It's unclear to me at the moment if this is associated with arrays of size-zero along certain dimensions. I also added new test files [here](https://github.com/shufay/ipie/tree/fm_fix/test_files). To run, 
```
python run_afqmc.py ints_ghf_nu=0.333333_V0=0.00055124_eps=7_nelec=3_rs=142.636_ecut=0.0012_nbsf=37_tri_2d.h5
```

**Comment by fdmalone** [comment]
Looks like @ankit76 found the additional issue [here](https://github.com/linusjoonho/ipie/pull/235/commits/1a22ed37cfeb4d4550f2cc42a0f0fe1fb169914a)
